File: Trabalho Pratico - 01/java/Alteração Aleatoria/Alteracao.java
import java.util.*;

public class Alteracao{

	public static void main(String[] args){
		MyIO.setCharset("UTF-8");
		Random gerador = new Random();
		gerador.setSeed(4);

		String palavra = MyIO.readLine();
		while(!palavra.equals("FIM")){
			char firstLetter = (char) ('a' + (Math.abs(gerador.nextInt())) % 26);
			char secondLetter = (char) ('a' + (Math.abs(gerador.nextInt())) % 26);
			alteraString(palavra, firstLetter, secondLetter);
			palavra = MyIO.readLine();
		}		
	}

	static void alteraString(String palavra, char a, char b){
		String newPalavra = "";
		for(int i = 0; i < palavra.length(); i++){
			if(palavra.charAt(i) == a){
				newPalavra  += b;
			}
			else{
				newPalavra += palavra.charAt(i);
			}
		}
		MyIO.println(newPalavra);
	}

}

File: Trabalho Pratico - 01/c/Alteração Aleatoria/Alteracao.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <time.h>

void alteraString(char *palavra, char a, char b) {
    char newPalavra[1000];
    int i;

    for (i = 0; palavra[i] != '\0'; i++) {
        if (palavra[i] == a) {
            newPalavra[i] = b;
        } else {
            newPalavra[i] = palavra[i];
        }
    }
    newPalavra[i] = '\0';

    printf("%s\n", newPalavra);
}

int main() {
    srand(4);

    char palavra[1000];

    while (1) {
        scanf("%s", palavra);
        if (strcmp(palavra, "FIM") == 0) {
            break;
        }

        char firstLetter = 'a' + (rand() % 26);
        char secondLetter = 'a' + (rand() % 26);
        alteraString(palavra, firstLetter, secondLetter);
    }

    return 0;
}
